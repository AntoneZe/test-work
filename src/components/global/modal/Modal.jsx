import React from "react";
import PropTypes from "prop-types";

import "./modal.less";

const Modal = ({ active, setActive, children }) => {
  return (
    <>
      <div
        className={active ? "custom-modal active" : "custom-modal"}
        onClick={() => setActive(false)}
      >
        <div
          className={
            active ? "custom-modal__content active" : "custom-modal__content"
          }
          onClick={(e) => e.stopPropagation()}
        >
          {children}
          <button
            onClick={() => setActive(false)}
            className={
              active
                ? "custom-modal__close-button active"
                : "custom-modal__close-button"
            }
          ></button>
        </div>
      </div>
    </>
  );
};

Modal.propTypes = {
  active: PropTypes.bool.isRequired,
  setActive: PropTypes.func.isRequired,
};

export default Modal;
