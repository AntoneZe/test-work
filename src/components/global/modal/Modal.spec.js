import React from "react";

import Modal from "./Modal";

describe("render Modal component", () => {
  it("should render Modal component", () => {
    const component = render(<Modal active={true} setActive={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});

describe("handle setActive callback", () => {
  it("should handle setActive callback", () => {
    const setActive = jest.fn();
    const handleClick = jest.spyOn(React, "useState");
    handleClick.mockImplementation((active) => [active, setActive]);

    const component = mount(<Modal active={true} setActive={setActive} />);
    component.find(".custom-modal").simulate("click");
    expect(setActive.mock.calls.length).toEqual(1);
  });

  it("should close modal", () => {
    const setActive = jest.fn();
    const handleClick = jest.spyOn(React, "useState");
    handleClick.mockImplementation((active) => [active, setActive]);

    const component = mount(<Modal active={true} setActive={setActive} />);
    component.find("button").simulate("click");
    expect(setActive).toBeCalledWith(false);
  });
});

describe("check modal className", () => {
  it("should have 'active' class", () => {
    const component = mount(<Modal active={true} setActive={() => {}} />);
    const modal = component.find(".custom-modal");
    expect(modal.hasClass("active")).toBe(true);
  });
  it("should't have 'active' class", () => {
    const component = mount(<Modal active={false} setActive={() => {}} />);
    const modal = component.find(".custom-modal");
    expect(modal.hasClass("active")).toBe(false);
  });
});
