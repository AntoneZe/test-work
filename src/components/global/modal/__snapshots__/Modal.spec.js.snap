// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Modal component should render Modal component 1`] = `
<div
  class="custom-modal active"
>
  <div
    class="custom-modal__content active"
  >
    <button
      class="custom-modal__close-button active"
    />
  </div>
</div>
`;
